homeassistant/components/camera/zoneminder.py

```diff
@@ -6,73 +6,32 @@
 """
 import asyncio
 import logging
-from urllib.parse import urljoin, urlencode
 
 from homeassistant.const import CONF_NAME
 from homeassistant.components.camera.mjpeg import (
     CONF_MJPEG_URL, CONF_STILL_IMAGE_URL, MjpegCamera)
-
-from homeassistant.components import zoneminder
+from homeassistant.components.zoneminder import DOMAIN
 
 _LOGGER = logging.getLogger(__name__)
 
 DEPENDENCIES = ['zoneminder']
-DOMAIN = 'zoneminder'
-
-# From ZoneMinder's web/includes/config.php.in
-ZM_STATE_ALARM = "2"
-
-
-def _get_image_url(hass, monitor, mode):
-    zm_data = hass.data[DOMAIN]
-    query = urlencode({
-        'mode': mode,
-        'buffer': monitor['StreamReplayBuffer'],
-        'monitor': monitor['Id'],
-    })
-    url = '{zms_url}?{query}'.format(
-        zms_url=urljoin(zm_data['server_origin'], zm_data['path_zms']),
-        query=query,
-    )
-    _LOGGER.debug('Monitor %s %s URL (without auth): %s',
-                  monitor['Id'], mode, url)
-
-    if not zm_data['username']:
-        return url
-
-    url += '&user={:s}'.format(zm_data['username'])
-
-    if not zm_data['password']:
-        return url
-
-    return url + '&pass={:s}'.format(zm_data['password'])
 
 
 @asyncio.coroutine
 def async_setup_platform(hass, config, async_add_entities,
                          discovery_info=None):
     """Set up the ZoneMinder cameras."""
-    cameras = []
-    monitors = zoneminder.get_state('api/monitors.json')
+    zm = hass.data[DOMAIN]
+
+    monitors = zm.get_monitors()
```

Comment:
Is this async safe, ie no I/O?

Reply:
Nope, it's directly hitting a ZM API and getting the values from there. This should be swapped to not be async.